File: ruby/sort/insertion_sort.rb
def insertion_sort(a)
  1.upto(a.size - 1) do |i|
    v = a[i]
    while i > 0 && a[i - 1] > v
      a[i] = a[i - 1]
      i -= 1
    end
    a[i] = v
  end
  a
end

File: python/sort/insertion_sort.py
def insertion_sort(a):
    for i in xrange(len(a)):
        v, j = a[i], i
        while j > 0 and a[j - 1] > v:
            a[j] = a[j - 1]
            j -= 1
        a[j] = v
    return a
